### ISSUE TITLE
Parser problem?

### ISSUE THREAD

**Issue by PaulBlythe** [microsoft/DirectXShaderCompiler#2779]
Not sure what to do about this.

A DX11 shader I am trying to convert has a variable in the pixel shader defined as 

`float4 Color0:COLOR0;`

This produces the error

` error: invalid usage of semantic 'COLOR0' in shader profile ps float4 Color0 : COLOR0;`

And it points at the variable name rather than the semantic name.

So I think the parser is using case insensitive comparisons for semantics and getting confused.

Are there any work arounds for this as the graphics guys have used this as a "standard" and I have several thousand cases of this to deal with .

Cheers

Paul

**Comment by siliconvoodoo** [comment]
I can't reproduce your problem. I think it works, it's even the case of the default snippet of code in Tim Jones's playground. `float4 color : COLOR;`

**Comment by PaulBlythe** [comment]
I solved it for me by changing all cases of COLOR0 in pixel shaders with SV_???

I ended up having to drop DXC and use glslc as DXC went into a an infinite loop on one shader.

Sadly this was in the middle of a multi-threaded shader crunch of over 8000 shaders and I couldn't identify which shader was causing the problem. The thread was locked so I couldn't get access to stdout to print debug info.

Thanks anyway
